"""tentando associar foreign key.

Revision ID: 5f17eb56a655
Revises: 3c735be6be6e
Create Date: 2022-06-29 14:24:05.537140

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '5f17eb56a655'
down_revision = '3c735be6be6e'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('user_control',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=20), nullable=True),
    sa.Column('nome', sa.String(length=80), nullable=True),
    sa.Column('senha', sa.String(length=14), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.drop_table('usuarios_control')
    op.drop_constraint('gastos_usuario_fkey', 'gastos', type_='foreignkey')
    op.create_foreign_key(None, 'gastos', 'user_control', ['usuario'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'gastos', type_='foreignkey')
    op.create_foreign_key('gastos_usuario_fkey', 'gastos', 'usuarios_control', ['usuario'], ['id'])
    op.create_table('usuarios_control',
    sa.Column('id', sa.INTEGER(), autoincrement=True, nullable=False),
    sa.Column('username', sa.VARCHAR(length=20), autoincrement=False, nullable=True),
    sa.Column('nome', sa.VARCHAR(length=80), autoincrement=False, nullable=True),
    sa.Column('senha', sa.VARCHAR(length=14), autoincrement=False, nullable=True),
    sa.PrimaryKeyConstraint('id', name='usuarios_control_pkey')
    )
    op.drop_table('user_control')
    # ### end Alembic commands ###
